src/cljs/athens/parse_renderer.cljs

```diff
@@ -94,13 +94,16 @@
      :component     (fn [& contents]
                       (components/render-component (first contents) uid))
      :page-link     (fn [& title] (render-page-link title))
-     :block-ref     (fn [uid]
-                      (let [block (pull db/dsdb '[*] [:block/uid uid])]
-                        [:span (use-style block-ref {:class "block-ref"})
-                         [:span {:class "contents" :on-click #(navigate-uid uid)}
-                          (if (= uid (:block/uid @block))
-                            [parse-and-render "{{SELF}}"]
-                            [parse-and-render (:block/string @block) uid])]]))
+     :block-ref     (fn [ref-uid]
+                      (let [block (pull db/dsdb '[*] [:block/uid ref-uid])]
+                        (if @block
+                          [:span (use-style block-ref {:class "block-ref"})
+                           [:span {:class "contents" :on-click #(navigate-uid ref-uid)}
+                            (if (= uid ref-uid)
+                              [parse-and-render "{{SELF}}"]
```

Comment:
This is actually slightly different behavior from Roam. Roam shows the content before and after the self-reference as well, but that's not that important

![image](https://user-images.githubusercontent.com/8952138/91784065-e884db00-ebcf-11ea-9a8f-e9d6922c336a.png)